src/compiler/parser.ts

```diff
@@ -8888,12 +8888,12 @@ namespace Parser {
                                 pushComment(scanner.getTokenText());
                             }
                             else {
-                                const whitespace = scanner.getTokenText();
+                                const whitespaceLength = scanner.getTextPos() - scanner.getTokenPos();
                                 // if the whitespace crosses the margin, take only the whitespace that passes the margin
-                                if (margin !== undefined && indent + whitespace.length > margin) {
-                                    comments.push(whitespace.slice(margin - indent));
+                                if (margin !== undefined && indent + whitespaceLength > margin) {
+                                    comments.push(scanner.getTokenText().slice(margin - indent));

```

Comment:
Well, I would've said not to do the double-`substring`/`slice` - but at this point I'm skeptical that it will affect perf much.

By the way, what is this logic trying to model?

Reply:
When you write

```js
/**
 * Some comments here
 *     Blockquote here
 *     And here
 * Back to original indentation
 */
```

The two blockquoted lines need to include 4 of the 5 spaces after the asterisk. Conversely, the first and last lines should skip the 1 space after the asterisk.

Needing to slice additional whitespace for the margin is uncommon, so putting the slice only in that branch should have improved performance noticeably. I'm not sure why it didn't.